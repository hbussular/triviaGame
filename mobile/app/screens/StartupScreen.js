import React from 'react'
import PropTypes from 'prop-types'
import { View } from 'react-native'

import ScreenProvider from '../containers/ScreenProvider'
import PrimaryButton from '../components/PrimaryButton'
import Logo from '../components/Logo'

import styles from './StartupScreen.style'

const StartupScreen = ({ navigation }) => (
  <ScreenProvider>
    <View style={styles.viewContainer}>
      <Logo />
      <PrimaryButton
        label="Get started"
        onPress={() => navigation.replace('QuestionScreen')}
      />
    </View>
  </ScreenProvider>
)

StartupScreen.propTypes = {
  navigation: PropTypes.object.isRequired,
}

export default StartupScreen
