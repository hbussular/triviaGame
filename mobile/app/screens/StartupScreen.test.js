import React from 'react'
import renderer from 'react-test-renderer'
import StartupScreen from './StartupScreen'

import { screenScenario } from '../tests/scenarios'

test('renders correctly', () => {
  const tree = renderer
    .create(<StartupScreen {...screenScenario} />)
    .toJSON()

  expect(tree).toMatchSnapshot()
})
