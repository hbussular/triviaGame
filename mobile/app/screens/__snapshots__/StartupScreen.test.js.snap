// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgba(213, 55, 43, 1)",
      "height": "100%",
      "padding": 20,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <Image
      collapsable={undefined}
      resizeMode="contain"
      source={1}
      style={
        Object {
          "height": "30%",
          "transform": Array [
            Object {
              "rotate": "0deg",
            },
          ],
          "width": "30%",
        }
      }
    />
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "rgba(255, 255, 255, 1)",
          "borderRadius": 8,
          "display": "flex",
          "opacity": 1,
          "paddingVertical": 20,
          "width": "100%",
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontWeight": "700",
              "textAlign": "center",
            },
            Object {
              "color": "rgba(213, 55, 43, 1)",
            },
            Object {
              "fontSize": 16,
            },
          ]
        }
      >
        Get started
      </Text>
    </View>
  </View>
</View>
`;
